### PR TITLE
06304 - Updates Metrics to use non-contiguous NodeIds

### LINKED ISSUE #6304: Noncontiguous NodeId Batch - All Metrics
Use AddressBook NodeId API

For each batch, these are the changes that need to be made. 
- NodeId - All methods returning a long node id now return NodeId
- NodeId - All methods consuming a long node id now consume NodeId
- NodeId - Change usage of int node ids as index of arrays/lists to  Map<NodeId, ?>
- NodeId - Complete translation of all other usage of int/long node ids to NodeId
- Ensure Unit Tests work with NonContiguous NodeIds.  

Files
- NoOpMetrics
- ConsensusMetricImpl
- NetworkMetrics
- PlatformMetrics
- IssMetrics
- Unit Tests (Related)

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:
Fixes #6304

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java [hunk 1]
```diff
@@ -177,17 +196,22 @@ public void update() {
...
-    public List<RunningAverageMetric> getAvgPingMilliseconds() {
-        return avgPingMilliseconds;
+    /**
+     * Returns the average number of bytes sent per second to each member.
+     *
+     * @return the average number of bytes sent per second to each member
+     */
+    @NonNull
+    public Map<NodeId, RunningAverageMetric> getAvgPingMilliseconds() {
+        return new HashMap<>(avgPingMilliseconds);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java:214]: Would it make sense to return an unmodifiable view of this map, rather than making a copy each time?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java [hunk 2]
```diff
@@ -196,8 +220,8 @@ public List<RunningAverageMetric> getAvgPingMilliseconds() {
      * @param connection the connection that was closed.
      */
     public void recordDisconnect(final Connection connection) {
-        int otherId = connection.getOtherId().getIdAsInt();
-        if (otherId >= 0 && otherId < disconnectFrequency.size()) {
+        NodeId otherId = connection.getOtherId();
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java:223]: can be `final`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java [hunk 3]
```diff
@@ -177,17 +196,22 @@ public void update() {
...
      * @return the average times, for each member, in milliseconds
      */
-    public double[] getPingMilliseconds() {
-        final double[] times = new double[avgPingMilliseconds.size()];
-        for (int i = 0; i < times.length; i++) {
-            times[i] = avgPingMilliseconds.get(i).get();
-        }
-        times[selfId.getIdAsInt()] = 0;
+    @NonNull
+    public Map<NodeId, Double> getPingMilliseconds() {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java:200]: This method could be replaced with a new `private` method, which calculates `pingValue` without creating a new map each time. Then this method could be deprecated

Up to you whether this is outside the scope of this issue

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkStatsTransmitter.java [hunk 4]
```diff
@@ -79,16 +83,24 @@ public void stop() {
     private void transmitStats() {
         // Send a transaction giving the average ping time from self to all others (in microseconds).
         // This data will eventually be used by chatter to optimize broadcast trees when they are implemented.
-        final int n = networkMetrics.getAvgPingMilliseconds().size();
-        final int[] avgPingMilliseconds = new int[n];
-        for (int i = 0; i < n; i++) {
-            avgPingMilliseconds[i] =
-                    (int) (networkMetrics.getAvgPingMilliseconds().get(i).get() * MILLISECONDS_TO_SECONDS);
-        }
+        Map<NodeId, RunningAverageMetric> nodePingAvgs = networkMetrics.getAvgPingMilliseconds();
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkStatsTransmitter.java:86]: `final`

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java:214]: You are right.   I think I was thinking about capturing the contents of the map in an unchanging state.  But the object references to the metrics don't change.  It's their internal state that is being updated.   Returning the map of metrics breaks the encapsulation somewhat.   Someone could spuriously update the metric.   The code path getting the map is marked for deletion, but I am going to change the method being called to return Map<NodeId, Double> to not leak the metrics references.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java:223]: made final.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkStatsTransmitter.java (hunk 4 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkStatsTransmitter.java:86]: made final.

**Review by alittley — APPROVED** [review]
